### PR TITLE
ci: enable production integration testing

### PR DESCRIPTION
**Pull request by wlee221**
_Issue #, if available:_

_Description of changes:_ This PR updates integration tests to run on both dev builds and prod builds. This also resolves sideEffects regression in `aws-amplify-react` that was found along the way.

I considered creating a separate job for prod testing, but this was inefficient because we would have to run `persist_to_workspace` for each of our sample to transition from dev to prod. 

I've tested that all our browser integration tests pass on both prod and dev with the accompanying PR on samples repo. See [link](https://app.circleci.com/pipelines/github/aws-amplify/amplify-js/6281/workflows/197a14ea-5ea6-466b-ad02-e4045e05d9b9) to see what this looks like on circleCI!

Screen Shot 2020-09-29 at 10 28 47 AM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/package.json [hunk 1]
```diff
@@ -1,108 +1,120 @@
 {
-  "name": "aws-amplify-react",
```
Comment (wlee221) [inline packages/aws-amplify-react/package.json:2]: Changed spaces to tabs per our `.prettierrc` config while I was here. Turn on "Hide whitespace changes" :D

File: packages/aws-amplify-react/package.json [hunk 2]
```diff
@@ -1,108 +1,120 @@
...
-    "@aws-amplify/ui": "^2.0.0",
-    "@aws-amplify/xr": "^2.0.0"
-  }
+	"name": "aws-amplify-react",
+	"version": "4.2.6",
+	"description": "AWS Amplify is a JavaScript library for Frontend and mobile developers building cloud-enabled applications.",
+	"main": "./lib/index.js",
+	"module": "./lib-esm/index.js",
+	"typings": "./lib-esm/index.d.ts",
+	"sideEffects": [
```
Comment (ashika01) [inline packages/aws-amplify-react/package.json:8]: how does this impact our package size? Can you add info on that?

Comment (wlee221) [reply packages/aws-amplify-react/package.json:8]: yes give me a second!

**Comment by wlee221** [comment]
@ashika01 
```
File sizes after gzip:

  117.26 KB (-58.61 KB)  build/static/js/2.ac17f930.chunk.js
  65.41 KB (+64.36 KB)   build/static/js/main.cfbc14b9.chunk.js
  3.62 KB                build/static/css/2.e5012852.chunk.css
  762 B                  build/static/js/runtime~main.a8a9905a.js
  514 B                  build/static/css/main.38b8828e.chunk.css
```

**Review by iartemiev — APPROVED** [review]
Looks good William! 👍🏻  Thanks for including the CCI pipeline run

File: packages/aws-amplify-react/package.json (hunk 2 above)
Comment (sammartinez) [reply packages/aws-amplify-react/package.json:8]: I would like to know as well.

**Review by ashika01 — APPROVED** [review]
Lets get this merged.

**Review by sammartinez — APPROVED** [review]
LGTM and im OK with the bundle size increase. Thanks for this @wlee221 !

File: packages/aws-amplify-react/package.json [hunk 3]
```diff
@@ -1,108 +1,120 @@
...
+		"./src/Storage/index.ts",
+		"./src/Storage/S3Album.ts",
```
Comment (ericclemmons) [inline packages/aws-amplify-react/package.json:10]: How are these considered side-effects?

https://github.com/aws-amplify/amplify-js/blob/d2f4bca73efda7375cf5f33bcffef88938f6c803/packages/aws-amplify-react/src/Storage/index.tsx

**Comment by ericclemmons** [comment]
Is there a benefit to running tests in both `dev` & `prod` mode? In terms of a test, until we have dev-specific features, it seems like doubling up on CI resources when `prod` is what matters.

File: packages/aws-amplify-react/package.json (hunk 3 above)
Comment (wlee221) [reply packages/aws-amplify-react/package.json:10]: To be honest, I'm not sure :/  But I confirmed manually that all three `./src/index.ts`, `./src/Storage/index.ts`, and `./src/Storage/S3Album.ts` need to be marked `sideEffects` or modules fail to load. My guess is that the app is expecting `S3Album`, and all files leading up to it (`index.ts` -> `Storage/index.ts` -> `src/Storage/S3Album.ts`) cannot not be swept.

**Comment by wlee221** [comment]
With our discussion today, I'll go ahead and merge this. Thanks everyone 👍

**Pull request merged**